nc/source/physics/collision/Simplex.h

```diff
@@ -0,0 +1,148 @@
+#pragma once
+
+#include "math/Vector.h"
+
+#include <array>
+
+namespace nc::physics
+{
+    class Simplex
+    {
+        public:
+            Simplex()
+                : m_points(),
+                  m_worldSupports(),
+                  m_localSupports(),
+                  m_size{0}
+            {}
+
+            template<size_t V1>
+            void ToPoint();
+
+            template<size_t V1, size_t V2>
+            void ToLine();
+
+            template<size_t V1, size_t V2, size_t V3>
+            void ToTriangle();
+
+            size_t size() const { return m_size; }
+            auto PointsBegin() const { return m_points.begin(); }
+            auto PointsEnd() const { return m_points.end(); }
+    
+            auto WorldSupportsBegin() const { return m_worldSupports.begin(); }
+            auto WorldSupportsEnd() const { return m_worldSupports.end(); }
+
+            auto LocalSupportsBegin() const { return m_localSupports.begin(); }
+            auto LocalSupportsEnd() const { return m_localSupports.end(); }
+
+            Vector3& operator[](size_t i) { return m_points[i]; }
+
+            void push_front(const Vector3& vertex, const Vector3& worldSupportA, const Vector3& worldSupportB, const Vector3& localSupportA, const Vector3& localSupportB)
```

Comment:
Why this casing?

Reply:
This guy was initially a wrapper around array, so I kept the same format. Doesn't really make sense anymore.